src/documents/tutorial-creatingAPreferencesEditorUsingThePreferencesFramework/PrefsFramework.md

```diff
@@ -0,0 +1,775 @@
+
+### Example code ###
+The source code used in this tutorial is one of the examples provided in the Infusion code base,
+which you can download from github: https://github.com/fluid-project/infusion. You’ll find the code
+for the simple Preference Editor in the examples folder:
+https://github.com/fluid-project/infusion/tree/master/examples/framework/preferences/minimalEditor.
+We recommend you download the Infusion library and load the example code into your favourite editor.
+<figure>
+![the folder hierarchy of the samepl code](../images/prefsEditorFolders.png)
+<figcaption>Figure 1: Folder hierarchy for the Preference Editor example</figcaption>
+</figure>
+
+The example code is a Preference Editor for the world’s first flying car.
+If you run a local webserver
+(for example using [this approach](http://www.linuxjournal.com/content/tech-tip-really-simple-http-server-python),
+or using [MAMP](https://www.mamp.info/en/))
+and navigate to the `index.html` file in a browser, you should see this interface:
+<figure>
+![The screen of the example Preference Editor](../images/simplePrefsEditor.png)
+<figcaption>Figure 2: The screen of the example Preference Editor</figcaption>
+</figure>
+
+This Preference Editor has only one preference – a simple boolean setting for the car’s auto-pilot
+feature – and a ‘save’ button. Try it out: If you check the checkbox and click save, the state of
+the preference will be stored in a cookie, and when you reload the page, the checkbox will be set
+to the saved value. Go ahead, try it.
+
+Let’s talk about what we’re seeing in this interface:
+<figure>
+![The parts of a Preference Editor screen](../images/prefsEditorParts.png)
+<figcaption>Figure 3: The parts of a Preference Editor screen</figcaption>
+</figure>
+
+* This Editor is a full-page editor, so all of what you see is Preferences Editor
+(the outer, blue dashed line).
+* The content inside the rectangle (outlined with a green dashed line) is called a Panel –
+a container representing one (or more) preferences. This particular Editor has only one Panel,
+but a realistic Editor will likely have several. This tutorial will teach you what you need to
+know to add more Panels to this Editor.
+* Inside the Panel is an Adjuster (outlined by the inner-most, orange dashed line) – the controls
+for adjusting a particular preference. This Panel has only one Adjuster in it, but you might want
+to create a Panel that has multiple Adjusters, say, in the case of very closely-related preferences.
+This tutorial will teach you about creating Panels with multiple Adjusters.
+
+Let’s take a close look at the code.
+
+#### Primary Schema ####
+The [Primary Schema](../PrimarySchemaForPreferencesFramework.md) is a document that defines the preferences for the Editor.
+The Primary Schema for our example Editor is defined in the
+`schemas/primary.js` file using the JSON format
+(you can learn about JSON at http://json.org/):
```

Comment:
It's not just in JSON - it is actually a document conforming to JSON Schema which should be referenced here: http://json-schema.org/